### PR TITLE
LAU-325: Add Staging DB for Functional Tests

### PR DESCRIPTION
**Pull request by davejones74**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-325

### Change description ###
Add ccd-data-store-api-staging database for running functional tests by the case-disposer.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: charts/ccd-case-disposer/values.preview.template.yaml [hunk 1]
```diff
@@ -101,7 +101,7 @@ ccd:
         DEFINITION_STORE_DB_HOST: ${SERVICE_NAME}-postgresql
         DEFINITION_STORE_IDAM_KEY: ${CCD_DEFINITION_STORE_S2S_SECRET}
         IDAM_USER_URL: https://idam-web-public.aat.platform.hmcts.net
-        DEFINITION_STORE_S2S_AUTHORISED_SERVICES: ccd_gw,ccd_gateway,fpl_case_service,ccd_data,ccd_ps,ccd_case_document_am_api,ccd_case_disposer
+        DATA_STORE_S2S_AUTHORISED_SERVICES: ccd_gw,ccd_gateway,fpl_case_service,ccd_data,ccd_ps,ccd_case_document_am_api
         OIDC_ISSUER: https://forgerock-am.service.core-compute-idam-aat2.internal:8443/openam/oauth2/hmcts
```
Comment (RababRabab) [inline charts/ccd-case-disposer/values.preview.template.yaml:105]: should this be in the DATA_STORE section?

File: charts/ccd-case-disposer/values.preview.template.yaml [hunk 2]
```diff
@@ -117,7 +117,7 @@ ccd:
       environment:
         DATA_STORE_DB_HOST: ${SERVICE_NAME}-postgresql
         DATA_STORE_IDAM_KEY: ${CCD_DATA_STORE_S2S_SECRET}
-        DATA_STORE_S2S_AUTHORISED_SERVICES: ccd_gw,ccd_gateway,fpl_case_service,ccd_data,ccd_ps
+        DEFINITION_STORE_S2S_AUTHORISED_SERVICES: ccd_gw,ccd_gateway,fpl_case_service,ccd_data,ccd_ps
         IDAM_USER_URL: https://idam-web-public.aat.platform.hmcts.net
```
Comment (RababRabab) [inline charts/ccd-case-disposer/values.preview.template.yaml:121]: should this be in the DEFINITION_STORE section?

**Review by RababRabab — APPROVED** [review]

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -1,17 +1,57 @@
...
+// Database
+////////////////////////////////
+
+variable "postgresql_user" {
+  default = "ccd"
+}
+
+variable "database_name" {
+  default = "ccd_data_store"
+}
```
[the remark this answers is not in the document]
Comment (davejones74) [reply infrastructure/variables.tf:41]: I think ccd_data_store as the database name is fine, it's the same one used in AAT. 'ccd_data' is one of the tables.

File: charts/ccd-case-disposer/values.preview.template.yaml [hunk 4]
```diff
@@ -14,7 +14,7 @@ elastic:
 
 elasticsearch:
   clusterName: "es"
-  imageTag: "7.17.1"
+  imageTag: "6.4.2"
```
Comment (danlysiak) [inline charts/ccd-case-disposer/values.preview.template.yaml:17]: I don't think this should be reverted? v7 is in Prod.

File: charts/ccd-case-disposer/values.preview.template.yaml (hunk 2 above)
Comment (danlysiak) [reply charts/ccd-case-disposer/values.preview.template.yaml:121]: And presumably needs the `ccd_case_disposer` to be an authorised service?

File: infrastructure/main.tf [hunk 5]
```diff
@@ -1,3 +1,111 @@
...
+  value        = random_string.draft_encryption_key.result
+  key_vault_id = data.azurerm_key_vault.ccd_shared_key_vault.id
+}
+
+resource "azurerm_key_vault_secret" "draft-store-key" {
+  name         = "${var.component}-draft-key"
+  value        = random_string.draft_encryption_key.result
+  key_vault_id = data.azurerm_key_vault.ccd_shared_key_vault.id
+}
+
```
Comment (danlysiak) [inline infrastructure/main.tf:111]: Do we need these?

File: infrastructure/main.tf [hunk 6]
```diff
@@ -1,3 +1,111 @@
...
+  is_aat     = "${(var.env == "aat" || var.env == "saat")}"
+  instance_count = "${local.is_aat ? 1 : 0}"
+
+  // Vault name
+  vaultName      = "${var.raw_product}-${var.env}"
+
+  // Shared Resource Group
+  sharedResourceGroup = "${var.raw_product}-shared-${var.env}"
+
+  sharedASPResourceGroup = "${var.raw_product}-shared-${var.env}"
```
Comment (danlysiak) [inline infrastructure/main.tf:19]: Looks redundant now (and has same value as `sharedResourceGroup` anyway).

File: infrastructure/main.tf [hunk 7]
```diff
@@ -1,3 +1,111 @@
...
+resource "random_string" "draft_encryption_key" {
+  length  = 16
+  special = true
+  upper   = true
+  lower   = true
+  number  = true
+  lifecycle {
+    ignore_changes = all
+  }
+}
```
Comment (danlysiak) [inline infrastructure/main.tf:36]: This also doesn't look required?

**Pull request merged**